android/src/main/java/com/swmansion/reanimated/layoutReanimation/AnimationsManager.java

```diff
@@ -376,6 +379,13 @@ public boolean isLayoutAnimationEnabled() {
 
   private boolean removeOrAnimateExitRecursive(View view, ViewGroup parent, boolean shouldRemove) {
     int tag = view.getId();
+    ViewManager viewManager = resolveViewManager(tag);
+
+    if (viewManager != null && viewManager.getName().equals("RNSScreenStack")) {
+      cancelAnimationsRecursive(view);
```

Comment:
Why we need to cancel animation instead of prevent animation start?

Reply:
There may already be some animated components in the screen.
If you don't cancel them now, when later their animations end, while traversing the tree up, they won't be able to cleanup their ancestors beyond the screen boundary, preventing those screens from being unmounted later (they'll have a non-zero `mExitingSubviewCountMap`)